internal/internal_workflow.go

```diff
@@ -664,6 +664,24 @@ func assertNotInReadOnlyState(ctx Context) {
 	}
 }
 
+func assertNotInReadOnlyStateCancellation(ctx Context) {
```

Comment:
One more question I forgot to ask before I mark approved. Can you think of any case where this can break a working workflow today? I can't, just checking.